### PR TITLE
PAYEE-767: Remove TaiResponse from TaxFreeAmountComparisonConnector.t…

### PR DESCRIPTION
**Pull request by Jake-Raffe**
…axFreeAmountComparison

- [ ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: app/uk/gov/hmrc/tai/connectors/TaxFreeAmountComparisonConnector.scala [hunk 1]
```diff
@@ -36,12 +35,12 @@ class TaxFreeAmountComparisonConnector @Inject()(val httpHandler: HttpHandler, s
...
-    ) recover {
+  def taxFreeAmountComparison(nino: Nino)(implicit hc: HeaderCarrier): Future[TaxFreeAmountComparison] =
+    httpHandler.getFromApiV2(taxFreeAmountComparisonUrl(nino.nino)).map { json =>
+      (json \ "data").as[TaxFreeAmountComparison]
+    } recover {
       case NonFatal(e) =>
-        logger.warn(s"Couldn't retrieve taxFreeAmountComparison for $nino with exception:${e.getMessage}")
-        TaiTaxAccountFailureResponse(e.getMessage)
+        logger.warn(s"Couldn't retrieve taxFreeAmountComparison for $nino with exception: ${e.getMessage}")
+        throw e
```
Comment (brendanstraughan) [inline app/uk/gov/hmrc/tai/connectors/TaxFreeAmountComparisonConnector.scala:44]: I know this is what we were doing before but it seems bad. I presume there is a recover higher up that will capture an exception?

Comment (sergi-hmrc) [reply app/uk/gov/hmrc/tai/connectors/TaxFreeAmountComparisonConnector.scala:44]: if the intention is to have a warn. I don't see it a bad practise (as long as we wanna warn here)

File: test/uk/gov/hmrc/tai/service/CodingComponentServiceSpec.scala [hunk 2]
```diff
@@ -98,36 +96,12 @@ class CodingComponentServiceSpec extends PlaySpec with MockitoSugar with FakeTai
...
-        val exceptionThrown = the[RuntimeException] thrownBy Await
-          .result(service.taxFreeAmountComparison(generateNino), 5.seconds)
-        exceptionThrown.getMessage must include("Help")
-      }
-
-      "a unknown exception is received" in {
-        val service = createSut
-
-        when(taxFreeAmountComparisonConnector.taxFreeAmountComparison(any())(any()))
-          .thenReturn(Future.successful(null))
```
[the remark this answers is not in the document]
Comment (sergi-hmrc) [reply test/uk/gov/hmrc/tai/service/CodingComponentServiceSpec.scala:124]: agree we should have the failure scenarios covered

**Review by PerrettJ4 — APPROVED** [review]
lgtm

**Review by brendanstraughan — APPROVED** [review]

**Review by sergi-hmrc — APPROVED** [review]

**Pull request merged**